app/assets/javascripts/discourse/controllers/reorder-categories.js.es6

```diff
@@ -0,0 +1,102 @@
+import ModalFunctionality from 'discourse/mixins/modal-functionality';
+import { bufferedProperty } from 'discourse/mixins/buffered-content';
+import binarySearch from 'discourse/lib/binary-search';
+import { popupAjaxError } from 'discourse/lib/ajax-error';
+
+// Modal for editing / creating a category
+export default Ember.Controller.extend(ModalFunctionality, Ember.Evented, {
+
+  categoriesBuffered: function() {
+    const bufProxy = Em.ObjectProxy.extend(BufferedProxy);
+    return this.get('site.categories').map(function(c) {
+      return bufProxy.create({ content: c });
+    });
+  }.property('site.categories'),
+
+  // uses propertyDidChange()
+  categoriesGrouped: function(key, newValue) {
+    if (arguments.length > 1) {
+      debugger;
+      return newValue;
+    }
+    const cats = this.get('categoriesBuffered');
+    const map = {};
+    cats.forEach(function(cat) {
+      const p = cat.get('position') || 0;
+      if (!map[p]) {
+        map[p] = {pos: p, cats: [cat]};
+      } else {
+        map[p].cats.push(cat);
+      }
+    });
+    const result = [];
+    Object.keys(map).map(p => parseInt(p)).sort((a,b) => a-b).forEach(function(pos) {
+      result.push(map[pos]);
+    });
+    return result;
+  }.property('categoriesBuffered'),
+
+  showApplyAll: function() {
+    let anyChanged = false;
+    this.get('categoriesBuffered').forEach(function(bc) {
+      anyChanged = anyChanged || bc.get('hasBufferedChanges');
+    });
+    return anyChanged;
+  }.property('categoriesBuffered.@each.hasBufferedChanges'),
+
+  saveDisabled: Em.computed.alias('showApplyAll'),
+
+  moveDir(cat, dir) {
+    const grouped = this.get('categoriesGrouped'),
+      curPos = cat.get('position'),
+      curGroupIdx = binarySearch(grouped, curPos, "pos"),
+      curGroup = grouped[curGroupIdx];
+
+    if (curGroup.cats.length === 1 && ((dir === -1 && curGroupIdx !== 0) || (dir === 1 && curGroupIdx !== (grouped.length - 1)))) {
+      const nextGroup = grouped[curGroupIdx + dir],
+        nextPos = nextGroup.pos;
+      cat.set('position', nextPos);
+    } else {
+      cat.set('position', curPos + dir);
+    }
+    cat.applyBufferedChanges();
+    Em.run.next(this, function() {
+      this.propertyDidChange('categoriesGrouped');
+      Em.run.schedule('afterRender', this, function() {
+        this.set('scrollIntoViewId', cat.get('id'));
+        this.trigger('scrollIntoView');
+      });
+    });
+  },
+
+  actions: {
+
+    moveUp: function(cat) {
+      this.moveDir(cat, -1);
+    },
+    moveDown: function(cat) {
+      this.moveDir(cat, 1);
+    },
+
+    commit: function() {
+      this.get('categoriesBuffered').forEach(function(bc) {
+        if (bc.get('hasBufferedChanges')) {
+          bc.applyBufferedChanges();
+          //bc.set('hasBufferedChanges', false);
```

Comment:
Nope, was leftover from when the `if` statement was checking the wrong property...